### DISCUSSION TITLE
Which strings in english.txt should be renamed or moved before 1.12?

### DISCUSSION THREAD

**Discussion by rubidium42** [OpenTTD/OpenTTD#9101] [General]
We've got english.txt which defines all the strings. The usage of some strings has changed over time, so their label/StringID and/or location in the file is not in line with the rest of the code. The goal of this discussion is to get a list of strings that should be renamed and potentially moved in the language files.
Renaming of strings must be done for all language files, and of course the StringIDs in the game need to be amended. When we do not rename the strings in the other languages, their translations will get lost causing unneeded extra work for the translators. This can likely be done by a relatively straight forward "sed" script over all the source files and then a review. Moving needs to happen manually in english.txt, though the web translator will then move the strings in the other languages.

This is explicitly not about the content of the strings themselves, just the defined labels/IDs. The aim is to perform all these renames and moves just "just" before the release, probably around the first beta or release candidate, so it does not create much extra work with backporting language updates made by the translators.

Let me start with:
| Old StringID | New StringID | New Block |
| --- | --- | --- |
|STR_GAME_OPTIONS_TOWN_NAMES_FRAME | STR_MAP_GEN_TOWN_NAME_LABEL | STR_MAP_GEN_* |
|STR_GAME_OPTIONS_TOWN_NAME_* | STR_MAP_GEN_TOWN_NAME_* | STR_MAP_GEN_* |
|STR_GAME_OPTIONS_ROAD_VEHICLES_DROPDOWN_* | STR_CONFIG_SETTING_ROAD_SIDE_* | Near STR_CONFIG_SETTING_ROAD_SIDE |

**Comment by frosch123** [comment]
| Old StringID | New StringID | New Block |
| --- | --- | --- |
| STR_DIFFICULTY_LEVEL_SETTING_MAXIMUM_NO_COMPETITORS | STR_AI_CONFIG_NUMBER | STR_AI_CONFIG_* |
| STR_CHEAT_SWITCH_CLIMATE_* | STR_CLIMATE_* | Near STR_TERRAIN_TYPE_VERY_FLAT |

**Comment by TrueBrain** [comment]
We have a lot of `STR_NNN_TRAIN_MMM` / `STR_NNN_ROAD_MMM` / .. combos, which are rather difficult for sorting, scripts to detect unused strings, etc. They are also often part of a list, and have to remain in that specific order.

I suggest we change them all to `STR_NNN_MMM_TRAIN` / `STR_NNN_MMM_ROAD`, etc. This is a long list, but to give a few examples:

```
STR_NEWS_FIRST_TRAIN_ARRIVAL                                    
STR_NEWS_FIRST_BUS_ARRIVAL                                      
...

STR_BUY_VEHICLE_TRAIN_LIST_TOOLTIP  
STR_BUY_VEHICLE_ROAD_VEHICLE_LIST_TOOLTIP 
```

You get the idea.

**Comment by TrueBrain** [comment]
| Old StringID | New StringID | New Block | Comment |
| --- | --- | --- | --- |
| STR_CONFIG_SETTING_PATHFINDER_YAPF_RECOMMENDED | STR_CONFIG_SETTING_PATHFINDER_YAPF |  | Prevents future problems when yayapf is added. |
